sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/service/schema/KeyspaceManager.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.stargate.sgv2.docsapi.service.schema;
+
+import io.opentelemetry.extension.annotations.WithSpan;
+import io.smallrye.mutiny.Multi;
+import io.smallrye.mutiny.Uni;
+import io.stargate.bridge.proto.QueryOuterClass;
+import io.stargate.bridge.proto.Schema;
+import io.stargate.bridge.proto.StargateBridge;
+import io.stargate.sgv2.common.cql.builder.Replication;
+import io.stargate.sgv2.docsapi.api.common.StargateRequestInfo;
+import io.stargate.sgv2.docsapi.api.exception.ErrorCode;
+import io.stargate.sgv2.docsapi.api.exception.ErrorCodeRuntimeException;
+import io.stargate.sgv2.docsapi.service.schema.common.SchemaManager;
+import io.stargate.sgv2.docsapi.service.schema.query.NamespaceQueryProvider;
+import java.util.function.Function;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+/** Keyspace manager provides basic operations on keyspaces. */
+@ApplicationScoped
+public class KeyspaceManager {
+
+  private static final Function<String, Uni<? extends Schema.CqlKeyspaceDescribe>>
+      MISSING_KEYSPACE_FUNCTION =
+          keyspace -> {
+            String message = "Namespace %s does not exist.".formatted(keyspace);
+            Exception exception =
+                new ErrorCodeRuntimeException(ErrorCode.DATASTORE_KEYSPACE_DOES_NOT_EXIST, message);
+            return Uni.createFrom().failure(exception);
+          };
+
+  @Inject SchemaManager schemaManager;
+
+  @Inject StargateRequestInfo requestInfo;
```

Comment:
Newbie question: KeyspaceManager is "@ApplicationScoped" which would suggest a singleton (one per-application), but `StargateRequestInfo` sounds like a per-request thing.
Wouldn't this mean that there's one instance created per request?

Reply:
Yes, `KeyspaceManager` is a singleton. But the `StargateRequestInfo` that gets injected into it is a proxy, so its behavior will change depending on the current request.

I found this the other day, quick read that gives a good overview of CDI in Quarkus: https://quarkus.io/guides/cdi

Reply:
Interesting. Did not think of that possibility. Thanks for the link!

Reply:
@olim7t to the rescue..